packages/fuels-programs/src/contract.rs

```diff
@@ -574,12 +615,27 @@ where
             .map_err(|err| map_revert_error(err, &self.log_decoder))
     }
 
-    pub async fn submit(mut self) -> Result<ContractCallHandler<T, D>> {
+    pub async fn submit(mut self) -> Result<SubmitResponse<T, D>> {
```

Comment:
If you go with the suggestion about send/submit in the provider, we should align this to say send as well